Code/Framework/AzCore/AzCore/DOM/DomBackendRegistryInterface.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/DOM/DomBackend.h>
+
+namespace AZ::DOM
+{
+    //! Central interface for registering and looking up backend types by file extension.
+    class BackendRegistryInterface
+    {
+    public:
+        AZ_RTTI(BackendRegistryInterface, "{B72A88AC-DF15-4F92-9489-ABCDEA3D94E6}")
+        using BackendFactory = AZStd::function<AZStd::unique_ptr<Backend>()>;
```

Comment:
This is also declared in `DomBackendRegistry.h`. Consider reducing this to a single declaration to avoid unexpected surprises when the function signature has to change.